ext/date/php_date.c

```diff
@@ -1143,7 +1143,7 @@ static zend_string *date_format(char *format, size_t format_len, timelib_time *t
 				length = slprintf(buffer, 32, "%02d", (int) isoweek); break; /* iso weeknr */
 			case 'o':
 				if(!weekYearSet) { timelib_isoweek_from_date(t->y, t->m, t->d, &isoweek, &isoyear); weekYearSet = 1; }
-				length = slprintf(buffer, 32, "%d", (int) isoyear); break; /* iso year */
+				length = slprintf(buffer, 32, "%lld", (zend_long) isoyear); break; /* iso year */
```

Comment:
I don't think you can just use ``%lld`` as I believe that doesn't work on Windows. PHP defines a few constants for this, such as ``ZEND_LONG_FMT_SPEC``, used somewhere else in this file: https://github.com/php/php-src/blob/master/ext/date/php_date.c#L4451

Reply:
@derickr I tested this on Windows - I actually *only* tested it on W10 x64 as it was the nearest working build env I had - and it works fine there, but it's possible it doesn't work on 32-bit Windows (which is exactly why I opened the PR rather than just merging, I felt this needed review from someone more experienced).

The choice of `%lld` was based on https://github.com/DaveRandom/php-src/blob/caed4b0387e95215817e6a5aea558bf6f985580a/ext/date/php_date.c#L1233, does this need migrating as well or is there something I'm missing?

Reply:
That is printing a timelib_sll, which uses `%lld`. Here you are using a `zend_long`, which uses `ZEND_LONG_FMT`.

Reply:
As both are, afaict, essentially aliases of the same type (64-bit signed integer) after a lot of tapdancing, would you mind explaining the functional difference here? I'm sure you're right, I'd just like to understand why

Reply:
timelib_sll is int64_t, zend_long is int64_t or int32_t, depending on platform.